libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/performance/KotlinPerformancePlugin.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.plugin.performance
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.execution.TaskExecutionListener
+import org.gradle.api.tasks.TaskState
+import org.jetbrains.kotlin.gradle.dsl.multiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.addExtension
+import java.io.File
+
+class TaskTimerListener : TaskExecutionListener {
+    companion object {
+        val tasksTimes = mutableMapOf<String, Double>()
+        fun getTime(taskName: String) = tasksTimes[taskName] ?: 0.0
+    }
+
+    private var startTime = System.nanoTime()
+
+    override fun beforeExecute(task: Task) {
+        startTime = System.nanoTime()
+    }
+
+    override fun afterExecute(task: Task, taskState: TaskState) {
+        tasksTimes[task.name] = (System.nanoTime() - startTime) / 1000.0
+    }
+}
+
+open class KotlinPerformancePlugin : Plugin<Project> {
+    private fun checkSettings(project: Project, performanceExtension: PerformanceExtension): Boolean {
+        var result = true
+        if (performanceExtension.metrics.isEmpty()) {
+            project.logger.warn("There is no tracked metrics. Please, provide metrics in settings of $EXTENSION_NAME plugin.")
+            result = false
+        }
+        // Find binaries with trackable performance.
+        if (performanceExtension.trackedBinaries.isEmpty()) {
+            project.logger.warn("There is no tracked binaries. Please, provide binaries in settings of $EXTENSION_NAME plugin.")
+            result = false
+        }
+        return result
+    }
+
+    private fun getCompilationResults(tasksNames: Iterable<String>, success: Boolean): String {
+        var time = 0.0
+        var status = true
+        tasksNames.forEach {
+            time += TaskTimerListener.getTime(it)
+            status = tasksNames.map { TaskTimerListener.tasksTimes.containsKey(it) }.reduce { a, b -> a && b }
```

Comment:
```kotlin
var status = success
....
status &= tasksNames.all { TaskTimerListener.tasksTimes.containsKey(it) }
```
?

Reply:
Do we really need to take into account compile time of all associated compilations?